public/app/plugins/datasource/cloudwatch/utils/logsRetry.test.ts

```diff
@@ -23,7 +27,11 @@ describe('runWithRetry', () => {
     queryFunc.mockReturnValueOnce(throwError(() => createErrorResponse(targets)));
     queryFunc.mockReturnValueOnce(of([createResponseFrame('A')]));
 
-    const valuesPromise = lastValueFrom(runWithRetry(queryFunc, targets).pipe(toArray()));
+    const startTime = new Date();
+    const timeoutFunc = () => {
+      return Date.now() >= startTime.valueOf() + 30000;
+    };
```

Comment:
you can move the timeoutFunc definition outside the test so you don't need to define it every time (you can have one function that returns true and one that returns false. Also, to avoid (unlikely) race conditions you can avoid real times and just mock true/false.